content/courses/advanced-ada/chapters/expressions.rst

```diff
@@ -3,6 +3,391 @@ Expressions
+       --        ^^^^^^^ membership choice list
+       --
+       --  ^^^^^^^^^^^^^ relation
+       --  ^^^^^^^^^^^^^ expression
+
+    end Show_Expression_Elements;
+
+In this code example, we see three expressions. As we mentioned earlier, every
+expression has a type; here, the type of each expression is :ada:`Boolean`.
+
+The first expression (:ada:`M1 in Active_Mode and then M2 in Off | A`) consists
+of two relations: :ada:`M1 in Active_Mode` and :ada:`M2 in Off | A`. Let's
+discuss some of the details.
+
+The :ada:`M1 in Active_Mode` relation consists of the simple expression
+:ada:`M1` and the membership choice list :ada:`Active_Mode`. (Here, the
+:ada:`in` keyword is part of the relation definition.) Also, as we see in the
+comments of the source code, the simple expression :ada:`M1` is, at the same
+time, a term, a factor, a primary and a name.
+
+Let's briefly talk about this chain of syntactic elements for simple
+expressions. Very roughly said, this is how we can break up simple expressions:
+
+- a simple expression consists of terms;
+
+- a term consists of factors;
+
+- a factor consists of primaries;
+
+- a primary can be one of those:
+
+    - a numeric literal;
+
+    - :ada:`null`;
+
+    - a string literal;
+
+    - :doc:`an aggregate <aggregates>`;
+
+    - a name;
+
+    - an allocator (like :ada:`new Integer`);
+
+    - :ref:`a parenthesized expression <Adv_Ada_Parenthesized_Expressions>`;
+
+    - :ref:`a conditional expression <Adv_Ada_Conditional_Expressions>`;
+
+    - :ref:`a quantified expression <Adv_Ada_Quantified_Expressions>`;
+
+    - :ref:`a declare expression <Adv_Ada_Declare_Expressions>`.
+
+.. admonition:: For further reading...
+
+    The definition of simple expressions we've just seen is very simplified. In
+    actuality, these are the grammatical elements specified in the Ada Reference
+    Manual:
+
+    .. code-block:: none
+
+        simple_expression ::=
+          [unary_adding_operator] term {binary_adding_operator term}
+
+        term ::= factor {multiplying_operator factor}
+
+        factor ::= primary [** primary] | abs primary | not primary
+
+        primary ::=
+          numeric_literal | null | string_literal | aggregate
+        | name | allocator | (expression)
+        | (conditional_expression) | (quantified_expression)
+        | (declare_expression)
+
+Later on in this chapter, we discuss
+:ref:`conditional expressions <Adv_Ada_Conditional_Expressions>`,
+:ref:`quantified expressions <Adv_Ada_Quantified_Expressions>` and
+:ref:`declare expressions <Adv_Ada_Declare_Expressions>` in more details.
+
+In the relation :ada:`M2 in Off | A` from the code example, :ada:`Off | A` is
+a membership choice list, and :ada:`Off` and :ada:`A` are membership choices.
+
+.. admonition:: For further reading...
+
+    Relations can actually be much more complicated than the one we just
+    seen. In fact, this is the definition from the Ada Reference Manual:
```

Comment:
"the one we just saw"

Reply:
Thanks! (I probably intended to write "we've just seen," but "we just saw" is OK.)